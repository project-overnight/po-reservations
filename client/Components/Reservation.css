*, *:before, *:after {
  box-sizing: border-box;
  font-family: Circular, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

.positioner {

}

.wrapper {
  position: absolute;
  top: 24px;
  margin-left: 45px;
  display: flex;
  --page-shell-max-content-width: 1440px;
  width: 100%;
  flex-direction: column;
  box-sizing: border-box;
}

.container {
  display: flex;
  align-self: flex-end;
  flex-direction: column;
  justify-content: space-around;
  padding: 8px 24px 24px;
  margin-top: 24px;
  box-sizing: border-box;
  border-radius: 12px;
  box-shadow: 0px 6px 16px rgba(0, 0, 0, 0.12);
  color: rgb(34, 34, 34);
  font-weight: 400;
  font-size: 16px;
  line-height: 20px;
}

.nocharge {
  display: flex;
  justify-content: center;
  font-size: 13px;
  line-height: 18px;
}

.list_item {
  display: flex;
  font-weight: 400;
}
