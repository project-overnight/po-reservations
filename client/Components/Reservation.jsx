import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Costs from './Costs';
import Form from './Form';
import TopRow from './TopRow';
import styles from './Reservation.css';
import { fetchData } from '../store';

const mapStateToProps = (state) => {
  const { checkin, checkout } = state.dateReducer;
  let duration = 0;
  if (checkin && checkout) {
    const checkinDate = checkin.slice(-2);
    const checkoutDate = checkout.slice(-2);
    duration = Number.parseInt(checkoutDate, 10) - Number.parseInt(checkinDate, 10);
  }
  const { room } = state.roomReducer;
  return { duration, room };
};

const mapDispatchToProps = {
  fetchData,
};

class ConnectedReservation extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    const { fetchData: fetch } = this.props;
    const params = new URLSearchParams(window.location.search);
    const record = Number.parseInt(params.toString(), 10);
    if (record) {
      fetch(record);
    } else {
      // console.error('No record associated with this URL.')
    }
  }

  handleSubmit(e) {
    e.preventDefault();
  }

  render() {
    const { duration, room } = this.props;
    return (
      <div className={styles.wrapper}>
        <div className={styles.container}>
          <TopRow data={room} />
          <Form
            data={room}
            duration={duration}
            handleSubmit={this.handleSubmit}
          />
          <div className={styles.nocharge}>
            { duration ? <p>You won&apos;t be charged yet</p> : ''}
          </div>
          <Costs data={room} duration={duration} />
        </div>
      </div>
    );
  }
}

ConnectedReservation.propTypes = {
  room: PropTypes.shape({
    price: PropTypes.number.isRequired,
    service: PropTypes.number.isRequired,
    cleaning: PropTypes.number.isRequired,
    tax: PropTypes.number.isRequired,
  }).isRequired,
  duration: PropTypes.number.isRequired,
  fetchData: PropTypes.func.isRequired,
};

const Reservation = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ConnectedReservation);

export default Reservation;
