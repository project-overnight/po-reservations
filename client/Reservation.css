html {
  font-family: Montserrat;
}

/* html {
  display: grid;
  grid-template-columns: 2fr 1fr;
  grid-template-rows: repeat(5, 1fr);
} */

.wrapper {
  display: flex;
  width: 100%;
  flex-direction: column;
}

.container {
  display: flex;
  align-self: flex-end;
  flex-direction: column;
  justify-content: space-around;
  padding: 1em;
  margin: 1em;
  border-radius: 9px;
  /* border: 1px solid black; */
  box-shadow: 0px 0px 0px 1px rgb(213, 213, 213);
}