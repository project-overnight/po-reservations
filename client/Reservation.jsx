import React, { Component } from 'react';
import axios from 'axios';
// eslint-disable-next-line no-unused-vars
import regeneratorRuntime from 'regenerator-runtime';
import styles from './Reservation.css';

class Reservation extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  async componentDidMount() {
    const data = await this.fetchData();
    this.setState(data);
  }

  // eslint-disable-next-line class-methods-use-this
  fetchData() {
    return axios.get('/api/room/1')
      .then((response) => {
        return response.data;
      })
      .catch((err) => console.log('Error downloading room:', err));
  }

  // eslint-disable-next-line class-methods-use-this
  handleSubmit(e) {
    e.preventDefault();
  }


  render() {
    const { price } = this.state;
    return (
      <div className={styles.wrapper}>
        <div className={styles.container}>
          <p id="price">{price}</p>
          <form id="form" onSubmit={this.handleSubmit}>
            <label htmlFor="checkin">
              Check-In
              <input type="date" placeholder="Add date" id="checkin" />
            </label>
            <label htmlFor="checkout">
              Checkout
              <input type="date" placeholder="Add date" id="checkout" />
            </label>
            <br />
            <label htmlFor="guests">
              Guests
              <input type="text" id="guests" placeholder="1 guest" />
            </label>
          </form>
          <button type="submit" form="form">Check availability</button>
        </div>
      </div>
    );
  }
}


// const Reservation = () => {
//   const [room, setRoom] = useState([]);

//   useEffect(() => {
//     };
//     fetchData();
//   }, []);

//   const handleSubmit = (e) => {
//     e.preventDefault();
//   };

//   return (
//     <div className={styles.wrapper}>
//       <div className={styles.container}>
//         <p>{room.price}</p>
//         <form id="form" onSubmit={handleSubmit}>
//           <label htmlFor="checkin">
//             Check-In
//             <input type="date" placeholder="Add date" id="checkin" />
//           </label>
//           <label htmlFor="checkout">
//             Checkout
//             <input type="date" placeholder="Add date" id="checkout" />
//           </label>
//           <br />
//           <label htmlFor="guests">
//             Guests
//             <input type="text" id="guests" placeholder="1 guest" />
//           </label>
//         </form>
//         <button type="submit" form="form">Check availability</button>
//       </div>
//     </div>
//   );
// };

export default Reservation;
