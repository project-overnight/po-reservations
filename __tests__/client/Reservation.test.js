/* eslint-disable no-console */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';

import Reservation from '../../client/Reservation.jsx';

describe('<Reservation /> sans prop information', () => {
  const shell = shallow(<Reservation />);
  // console.log(instance);
  // instance.handleSubmit = jest.fn();

  it('should be the same as the snapshot', () => {
    expect(shell.html()).toMatchSnapshot();
  });

  it('should have a Submit button', () => {
    expect(shell.find('button[type="submit"]').length).toBe(1);
  });

  it('should have two calendar inputs', () => {
    expect(shell.find('input[type="date"]').length).toBe(2);
  });

  it('should have three labels', () => {
    expect(shell.find('label').length).toBe(3);
  });
});
