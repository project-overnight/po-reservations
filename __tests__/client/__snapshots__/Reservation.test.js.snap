// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reservation /> sans prop information should be the same as the snapshot 1`] = `"<div class=\\"wrapper\\"><div class=\\"container\\"><p id=\\"price\\"></p><form id=\\"form\\"><label for=\\"checkin\\">Check-In<input type=\\"date\\" placeholder=\\"Add date\\" id=\\"checkin\\"/></label><label for=\\"checkout\\">Checkout<input type=\\"date\\" placeholder=\\"Add date\\" id=\\"checkout\\"/></label><br/><label for=\\"guests\\">Guests<input type=\\"text\\" id=\\"guests\\" placeholder=\\"1 guest\\"/></label></form><button type=\\"submit\\" form=\\"form\\">Check availability</button></div></div>"`;
